src/clusterfuzz/_internal/base/concurrency.py

```diff
@@ -21,11 +21,30 @@
 POOL_SIZE = multiprocessing.cpu_count()
 
 
+class InProcessPool:
```

Comment:
can you add docstrings? 

Also -- this is more of a "SingleThreadedPool" right?